bolt/src/main/java/com/slack/api/bolt/service/builtin/AmazonS3InstallationService.java

```diff
@@ -26,19 +28,38 @@
 public class AmazonS3InstallationService implements InstallationService {
 
     private final String bucketName;
-    private AwsCredentialsProvider credentialsProvider;
+
+    private final AwsCredentialsProvider credentialsProvider;
+    private final Region region;
+    private final URI endpointOverride;
+
     private boolean historicalDataEnabled;
 
     public AmazonS3InstallationService(String bucketName) {
+        this(bucketName, DefaultCredentialsProvider.create(), null, null);
+    }
+
+    public AmazonS3InstallationService(String bucketName, AwsCredentialsProvider credentialsProvider) {
+        this(bucketName, credentialsProvider, null, null);
+    }
+
+    public AmazonS3InstallationService(
+            String bucketName,
+            AwsCredentialsProvider credentialsProvider,
+            Region region,
+            String endpointOverride
+    ) {
         this.bucketName = bucketName;
+        this.credentialsProvider = credentialsProvider;
+        this.region = region != null || System.getenv("AWS_REGION") == null ? region : Region.of(System.getenv("AWS_REGION"));
```

Comment:
This might just be my Java getting rusty but I found this line hard to read, could we add brackets to make it more explicit

```suggestion
        this.region = (region != null || System.getenv("AWS_REGION") == null) ? region : Region.of(System.getenv("AWS_REGION"));
```

Reply:
I suggest making region is string and simplify:

```suggestion
        this.region = StringUtils.isBlank(region) ? Region.of(System.getenv("AWS_REGION")) : Region.of(region);
```

And let null checks happen in the aws sdk downstream.

Reply:
Region.of method throws a runtime exception if the `System.getenv("AWS_REGION")` is null. This situation can arise when you run the code outside of AWS infra. Thus, I'd like to avoid changing this logic so far.